official/vision/beta/projects/yolo/tasks/yolo.py

```diff
@@ -0,0 +1,442 @@
+
+    if not self.task_config.init_checkpoint:
+      logging.info("Training from Scratch.")
+      return
+
+    ckpt_dir_or_file = self.task_config.init_checkpoint
+    if tf.io.gfile.isdir(ckpt_dir_or_file):
+      ckpt_dir_or_file = tf.train.latest_checkpoint(ckpt_dir_or_file)
+
+    # Restoring checkpoint.
+    if self.task_config.init_checkpoint_modules == 'all':
+      ckpt = tf.train.Checkpoint(model=model)
+      status = ckpt.restore(ckpt_dir_or_file)
+      status.expect_partial().assert_existing_objects_matched()
+    elif self.task_config.init_checkpoint_modules == 'backbone':
+      ckpt = tf.train.Checkpoint(backbone=model.backbone)
+      status = ckpt.restore(ckpt_dir_or_file)
+      status.expect_partial().assert_existing_objects_matched()
+    elif self.task_config.init_checkpoint_modules == 'decoder':
+      ckpt = tf.train.Checkpoint(backbone=model.backbone, decoder=model.decoder)
+      status = ckpt.restore(ckpt_dir_or_file)
+      status.expect_partial() 
+    else:
+      assert "Only 'all' or 'backbone' can be used to initialize the model."
+
+    logging.info('Finished loading pretrained checkpoint from %s',
+                 ckpt_dir_or_file)
+
+  def _wrap_optimizer(self, optimizer, runtime_config):
+    """Wraps the optimizer object with the loss scale optimizer."""
+    if runtime_config and runtime_config.loss_scale:
+      use_float16 = runtime_config.mixed_precision_dtype == "float16"
+      optimizer = performance.configure_optimizer(
+          optimizer,
+          use_graph_rewrite=False,
+          use_float16=use_float16,
+          loss_scale=runtime_config.loss_scale)
+    return optimizer
+
+  def create_optimizer(self,
+                       optimizer_config: OptimizationConfig,
+                       runtime_config: Optional[RuntimeConfig] = None):
+    """Creates an TF optimizer from configurations.
+
+    Args:
+      optimizer_config: the parameters of the Optimization settings.
+      runtime_config: the parameters of the runtime.
+
+    Returns:
+      A tf.optimizers.Optimizer object.
+    """
+    opt_factory = optimization.YoloOptimizerFactory(optimizer_config)
+    ema = opt_factory._use_ema
+    opt_factory._use_ema = False
+
+    opt_type = opt_factory._optimizer_type
+    if (opt_type == 'sgd_torch'):
+      optimizer = opt_factory.build_optimizer(opt_factory.build_learning_rate())
+      optimizer.set_bias_lr(
+          opt_factory.get_bias_lr_schedule(self._task_config.smart_bias_lr))
+
+      weights, biases, others = self._model.get_weight_groups(
+          self._model.trainable_variables)
+      optimizer.set_params(weights, biases, others)
+    else:
+      optimizer = opt_factory.build_optimizer(opt_factory.build_learning_rate())
+    opt_factory._use_ema = ema
+
+    if ema:
+      logging.info("EMA is enabled.")
+    optimizer = opt_factory.add_ema(optimizer)
+    optimizer = self._wrap_optimizer(optimizer, runtime_config)
+    return optimizer
+
+
+class ListMetrics:
```

Comment:
Sorry, why do you need this class?

Reply:
this allows for a more readable log when training. this is because we log metrics for each path independently

Reply:
let me try that instead.